source/link/linker.cpp

```diff
@@ -186,7 +192,8 @@ spv_result_t Linker::Link(const uint32_t* const* binaries,
                                       SPV_ERROR_INVALID_BINARY)
            << "No modules were given.";
 
-  std::vector<std::unique_ptr<Module>> modules;
+  std::vector<std::unique_ptr<IRContext>> contexts;
+  std::vector<Module*> modules;
```

Comment:
Curious why did the vector of Modules change from a vector of unique_ptr to a vector of raw ptr?

Reply:
Convenience. The modules are owned by the vector<unique_ptr<IRContext>>, but most of the code referred to the module vector. So to avoid further rewrites I also tracked raw pointers to the modules to avoid larger changes in the linker.